packages/language-server/src/completion-items.ts

```diff
@@ -371,13 +371,34 @@ function rangeContains(range: Range, inner: Range): boolean {
   return atMost(range.start, inner.start) && atMost(inner.end, range.end);
 }
 
-function createInsertRange(line: number, column: number): Range {
+/**
+ * Create an insert range at the position right after the sent column.
+ * For exmaple, for text "12345" the character in position 1 is "1" and the
+ * insert position after it is "1|2345".
+ * The character in position 2 is "2" and the insert position after it is "12|345".
+ * The character at position 5 is "5" and the insert position after it at the end of the text -> "12345|".
+ * */
+function createInsertRangeAfter(line: number, column: number): Range {
   return {
     start: Position.create(line - 1, column),
     end: Position.create(line - 1, column),
   };
 }
 
+/**
+ * Create an insert range at the position right before the sent column.
+ * For exmaple, for text "12345" the character in position 1 is "1" and the
+ * insert position before it is the beginning of the text -> "|12345".
+ * The character in position 2 is "2" and the insert position before it is "1|2345".
+ * The character at position 5 is "5" and the insert position before it is "1234|5".
+ * */
+function createInsertRangeBefore(line: number, column: number): Range {
+  return {
+    start: Position.create(line - 1, column - 1),
```

Comment:
comment that Chevrotain line/columns are `1 based` while VSCode line/columns are 0 based?

Reply:
It's a bit more complicated than that... Chevrotain position is a position **of** a character while vscode position is a position **between** characters. That's why there's a need for `createInsertRangeBefore` and `createInsertRangeAfter`.

Reply:
even a better reason to comment this or maybe even extract some utility/function with documentation.